ubi_manifest/worker/tasks/depsolver/rpm_depsolver.py

```diff
@@ -45,54 +49,64 @@ def __init__(
 
-        self._provides: Set = set()  # set of all rpm.provides we've visited
-        self._requires: Set = set()  # set of all rpm.requires we've visited
+        self._provides: Set[RpmDependency] = set()  # set of rpm.provides we've visited
+        self._requires: Set[RpmDependency] = set()  # set of rpm.requires we've visited
 
         # set of solvables (pkg, lib, ...) that we use for checking remaining requires
-        self._unsolved: Set = set()
+        self._unsolved: Set[RpmDependency] = set()
 
-        # Set of all modular rpms. Modifying the given modular_rpm_filenames set in place.
-        self._modular_rpm_filenames: Set = modular_rpm_filenames
+        # Set of all modular rpms. Modifying the given modular_rpm_filenames set in place
+        self._modular_rpm_filenames: Set[str] = modular_rpm_filenames
 
-        self._executor: ThreadPoolExecutor = Executors.thread_pool(
+        self._executor: ThreadPoolExecutor = Executors.thread_pool(  # type: ignore [assignment]
             max_workers=MAX_WORKERS
         )
         self._base_pkgs_only = kwargs.get("base_pkgs_only") or False
 
-    def __enter__(self):
+    def __enter__(self) -> Depsolver:
         return self
 
-    def __exit__(self, *args, **kwargs):
+    def __exit__(self, *args: Any, **kwargs: Any) -> None:
         self._executor.__exit__(*args, **kwargs)
 
-    def _get_pkgs_from_all_modules(self, repos):
+    def _get_pkgs_from_all_modules(
+        self, repos: list[YumRepository]
+    ) -> Future[set[str]]:
         """
         Search for modulemds in all input repos and extract rpm filenames.
         """
 
-        def extract_modular_filenames():
+        def extract_modular_filenames() -> set[str]:
             filenames = set()
-            for module in modules:
+            for module in modules:  # type: ignore [attr-defined]
                 filenames |= set(module.artifacts_filenames)
 
             return filenames
 
         modules = search_modulemds([Criteria.true()], repos)
         return f_proxy(self._executor.submit(extract_modular_filenames))
 
-    def get_base_packages(self, repos, pkgs_list, blacklist):
+    def get_base_packages(
+        self,
+        repos: list[YumRepository],
+        pkgs_list: set[str],
+        blacklist: list[PackageToExclude],
+    ) -> list[UbiUnit]:
         crit = create_or_criteria(["name"], [(rpm,) for rpm in pkgs_list])
 
         content = f_proxy(
             self._executor.submit(search_rpms, crit, repos, BATCH_SIZE_RPM)
         )
+
         newest_rpms = get_n_latest_from_content(
-            content, blacklist, self._modular_rpm_filenames
+            content, blacklist, self._modular_rpm_filenames  # type: ignore [arg-type]
         )
 
         return newest_rpms
 
-    def get_modulemd_packages(self, repos, pkgs_list):
+    def get_modulemd_packages(
+        self, repos: list[YumRepository], pkgs_list: set[str]
+    ) -> Future[Future[set[UbiUnit]]]:
```

Comment:
I guess in future we could use custom types for more complex types, e.g. variant is available in py312
https://docs.python.org/3/reference/simple_stmts.html#type